src/Params/ReportMakeParams.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Avtocod\B2BApi\Params;
+
+class ReportMakeParams
+{
+    /**
+     * Unique report type ID (e.g.: `some_report_uid` or `some_report_uid@domain`).
+     *
+     * @var string
+     */
+    private $report_type_uid;
+
+    /**
+     * Request type (e.g.: `VIN`, `GRZ`, `STS`, `PTS`, `CHASSIS`, etc.).
+     *
+     * @var string
+     */
+    private $type;
+
+    /**
+     * Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ)).
+     *
+     * @var string
+     */
+    private $value;
+
+    /**
+     * Additional request options.
+     *
+     * @var array<mixed>|null
+     */
+    private $options = [];
+
+    /**
+     * Force update report, if it already was generated previously.
+     *
+     * @var bool
+     */
+    private $is_force = false;
+
+    /**
+     * URL to call (using `post` method) when report content updated.
+     *
+     * @var string|null
+     */
+    private $on_update_url;
+
+    /**
+     * URL to call (using `post` method) when report generation completed.
+     *
+     * @var string|null
+     */
+    private $on_complete_url;
+
+    /**
+     * Additional request data.
+     *
+     * @var array<mixed>|null
+     */
+    private $data;
+
+    /**
+     * Idempotence key which the server uses to recognize subsequent retries of the same request.
+     *
+     * @var string|null
+     */
+    private $idempotence_key;
+
+    /**
+     * ReportMakeParameter constructor.
+     *
+     * @param string $report_type_uid Unique report type ID (e.g.: `some_report_uid` or `some_report_uid@domain`)
+     * @param string $type            Request type (e.g.: `VIN`, `GRZ`, `STS`, `PTS`, `CHASSIS`, etc.)
+     * @param string $value           Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ))
+     */
+    public function __construct(string $report_type_uid, string $type, string $value)
+    {
+        $this->report_type_uid = $report_type_uid;
+        $this->type            = $type;
+        $this->value           = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getReportTypeUid(): string
+    {
+        return $this->report_type_uid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param array<mixed>|null $options
```

Comment:
`array<mixed>` is correct type? `array<string, string|int|float|bool|array<mixed>>` seems more strict and useful

Reply:
Yes, it is more strict. Fixed 8ef870ed4bfbf66da60f6bcf51ae6af40b319753

Reply:
apply this changes to everywhere, please

Reply:
I fixed doc-block for `options` and change `array<mixed>` to `array<string, string|int|float|bool|null|array<mixed>>` for `data`-property 69c8732eb9914b6ca269031c792b23ffb6c7ad85